core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -0,0 +1,100 @@
+package io.grpc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * TODO: Insert description here. (generated by notcarl)
+ */
+public final class MessageEncoding {
+
+  /**
+   * Special sentinel codec indicating that no compression should be used.  Users should use
+   * reference equality to see if compression is disabled.
+   */
+  public static final Codec NONE = new Codec() {
+    @Override
+    public InputStream decompress(InputStream is) throws IOException {
+      return is;
+    }
+
+    @Override
+    public String getMessageEncoding() {
+      return null;
+    }
+
+    @Override
+    public OutputStream compress(OutputStream os) throws IOException {
+      return os;
+    }
+  };
+
+
+  /**
+   * Represents a message compressor.
+   */
+  public interface Compressor {
```

Comment:
@nmittler Compressor and Decompressor are pretty tied to their message encoding name.  Both need their name in order to be self describing.

Codec is present out of convenience, not necessity.  The composition over inheritance argument doesn't apply here, since that is for implementation inheritance, not interface inheritance.  With the latest changes, Codec is currently only used in this class and not used elsewhere.  It is exposed here for client convenience.